### DISCUSSION TITLE
RFE: allow to "mock" variables

### DISCUSSION THREAD

**Discussion by cjeanner** [ansible/ansible-lint#3717] [Ideas]
Hello World!

While trying to get my code square and clean, I'm facing a "nasty" issue: our code is usually running in CI, namely Zuul. Zuul provides some parameters directly to the env, allowing to consume a mighty `zuul` mapping.

Of course, since it comes from an external service, ansible-lint doesn't know about that parameter, and crashes with `internal-error` since the parameter isn't known:
```Bash
bash-5.1# ansible-playbook -i localhost, --syntax-check ci/playbooks/content_provider/content_provider.yml
ERROR! 'zuul' is undefined. 'zuul' is undefined
```

A nice way to work around that specific issue would be to add a new configuration parameter, such as `mock_parameters`, in the same way we currently have the `mock_roles` and `mock_modules`.

We would then pass this to the ansible-lint configuration file:
```YAML
[...]
mock_modules:
  - zuul_return
mock_parameters:
  - zuul
```
allowing then to get things working as expected.

WDYT?

**Comment by MarkusTeufelberger** [comment]
Do you mean `facts` or `variables` instead of `parameter`? I would assume that a parameter is something you are handing over to `ansible-playbook` on the command line for example...

**Comment by ssbarnea** [comment]
Have you considered making the change in such way that ansible loads this variables? You could use inventory for that and achieve the same effect without having to add a new feature to the linter.

**Reply by MarkusTeufelberger** [reply]
That's how I currently set variables that are required to be present in `molecule` for example. For syntax checks it might be a bit overkill or maybe not very portable, but then again it would also not be very portable to define facts within the `.ansible-lint` config file.
